zhaquirks/schneiderelectric/smoke.py

```diff
@@ -0,0 +1,174 @@
+            id=0x0004,
+            type=t.uint8_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+        se_test_mode: Final = ZCLAttributeDef(
+            id=0x0005,
+            type=t.Bool,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+        se_no_disturb_mode: Final = ZCLAttributeDef(
+            id=0x0006,
+            type=t.Bool,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+        se_cluster_revision: Final = ZCLAttributeDef(
+            id=0xFFFD,
+            type=t.uint16_t,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+
+
+(
+    QuirkBuilder(SE_MANUF_NAME, "W599001")
+    .applies_to(SE_MANUF_NAME, "W599501")
+    .applies_to(SE_MANUF_NAME, "755WSA")
+    .replaces(SEBasic, endpoint_id=20)
+    .replaces(SEAlarmConfiguration, endpoint_id=20)
+    .enum(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_led_brightness.name,
+        translation_key="led_brightness",
+        fallback_name="LED brightness",
+        enum_class=SELEDBrightness,
+    )
+    .enum(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_alarm_sound_level.name,
+        translation_key="alarm_sound_level",
+        fallback_name="Alarm sound level",
+        enum_class=SEAlarmSoundLevel,
+    )
+    .enum(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_alarm_sound_mode.name,
+        translation_key="alarm_sound_mode",
+        fallback_name="Alarm sound mode",
+        enum_class=SEAlarmSoundMode,
+    )
+    .sensor(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_lifetime.name,
+        translation_key="lifetime",
+        fallback_name="Lifetime",
+        device_class=NumberDeviceClass.DURATION,
+        unit=UnitOfTime.YEARS,
+        multiplier=0.5,
+        entity_type=EntityType.DIAGNOSTIC,
+    )
+    .number(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_hush_duration.name,
+        translation_key="hush_duration",
+        fallback_name="Hush duration",
+        device_class=NumberDeviceClass.DURATION,
+        unit=UnitOfTime.MINUTES,
+        min_value=5,
+        max_value=15,
+        step=1,
+    )
+    .switch(
+        cluster_id=SEAlarmConfiguration.cluster_id,
+        endpoint_id=20,
+        attribute_name=SEAlarmConfiguration.AttributeDefs.se_test_mode.name,
+        translation_key="test_mode",
+        fallback_name="Test mode",
+        reporting_config=ReportingConfig(
+            min_interval=15,
+            max_interval=300,
```

Comment:
Do we need `max_interval=300`? That would mean this attribute is reported at least every 300 seconds.

I feel like we can set this to `0` (so no periodic reporting) or just increase it a lot more.
Probably fine for `min_interval` as well (minimum delay between reports), since this shouldn't change often.
So, only `reportable_change=1`, others 0. (Or increase max_interval a lot, e.g. to `10800` to make sure we eventually get the updated attribute, even if we missed a report once. This might also be nice for "do not disturb" mode, especially if they can also be toggled from the device.)

See [ZCL R8](https://zigbeealliance.org/wp-content/uploads/2021/10/07-5123-08-Zigbee-Cluster-Library.pdf) page 72 and 73 (line 2695) for more information on reporting config.

Reply:
Fixed